### PR TITLE
Add ability to change media on notes

### LINKED ISSUE #19: Implement ability to change media on notes

### PR DESCRIPTION
**Pull request by carinatam**

### CODE REVIEW

File: src/components/MediaHandler/MediaHandler.js [hunk 1]
```diff
@@ -16,6 +17,7 @@ export class MediaHandler extends React.Component {
       const location = await FileInterface.saveMedia();
       mediaChanged(location);
     } catch {}
+    this.setState({buttonText: "Change Media"});
```
Comment (GarvinH) [inline src/components/MediaHandler/MediaHandler.js]: I think we can remove this line? obtaining a media does not guarantee a render. Meaning that if it fails, it should say add media??? Not sure about this one

Comment (carinatam) [reply src/components/MediaHandler/MediaHandler.js]: Maybe it should be moved to inside the try statement?

Comment (GarvinH) [reply src/components/MediaHandler/MediaHandler.js]: there is a then block inside load media which updates the state if there is a (relatively) successful load of the file. you can put it in there

File: src/components/MediaHandler/MediaHandler.js [hunk 2]
```diff
@@ -51,11 +54,10 @@ export class MediaHandler extends React.Component {
...
         >
-          <button onClick={this.obtainMedia}>Add Media</button>
+          <button onClick={this.obtainMedia}>{buttonText}</button>
         </div>
       );
     } else {
-      const { mime, error } = this.state;
       const fileLocation = `file://${mediaLocation}`;
 
       try {
```
Comment (GarvinH) [inline src/components/MediaHandler/MediaHandler.js:62]: In retrospect, I think copying and pasting the div and button was a bad implementation.

can you reimplement it so that it renders the content dynamically?

for example, right before the try statement
```
const content = (() => {
if (_.isNil(mime)) {
return null
} else if (error) {
return <h4>Failed to read file</h4>
} else if (mime includes "image) {
return <img .... />
}
.
.
.
})()
```

then you can just render it as 
```
<div>
  {content}
  <button>
</div>
```

reminder to do this in a separate commit from the button text one

File: src/components/MediaHandler/MediaHandler.js (hunk 1 above)
Comment (carinatam) [reply src/components/MediaHandler/MediaHandler.js]: Turns out I had it in there as well.

File: src/components/MediaHandler/MediaHandler.js [hunk 3]
```diff
@@ -51,13 +53,47 @@ export class MediaHandler extends React.Component {
...
+            <source src={fileLocation} type={mime} />
+          </audio>
+        }
+        else if(_.includes(mime, "video")){
+          <video controls>
+            <source src={fileLocation} type={mime} />
+          </video>
+        }
+      });
+      <div
```
Comment (GarvinH) [inline src/components/MediaHandler/MediaHandler.js]: you'll want to move the div down and replace the big if statement inside the try.

make sure you return the div otherwise it won't render

File: src/components/MediaHandler/MediaHandler.js [hunk 4]
```diff
@@ -51,13 +53,47 @@ export class MediaHandler extends React.Component {
...
+          <audio controls>
+            <source src={fileLocation} type={mime} />
+          </audio>
+        }
+        else if(_.includes(mime, "video")){
+          <video controls>
+            <source src={fileLocation} type={mime} />
+          </video>
+        }
+      });
```
Comment (GarvinH) [inline src/components/MediaHandler/MediaHandler.js]: right now this is still a function, you'll want to invoke it immediately to get the value inside the content with()

File: src/components/MediaHandler/MediaHandler.js (hunk 3 above)
Comment (GarvinH) [reply src/components/MediaHandler/MediaHandler.js]: i forgot to make a correction, you'll want to move all this into the try statement. my bad

File: src/components/MediaHandler/MediaHandler.js [hunk 5]
```diff
@@ -51,82 +53,47 @@ export class MediaHandler extends React.Component {
...
         </div>
       );
     } else {
-      const { mime, error } = this.state;
       const fileLocation = `file://${mediaLocation}`;
-
-      try {
-        if (_.isNil(mime)) {
+      
+      const content = (() => {
```
Comment (GarvinH) [inline src/components/MediaHandler/MediaHandler.js]: i think it would be safer to move this into the try as well

**Review by GarvinH — APPROVED** [review]
👍
